uitools/cpp/Esri/ArcGISRuntime/Toolkit/BasemapGalleryController.cpp

```diff
@@ -508,16 +508,38 @@ namespace Toolkit {
    */
   void BasemapGalleryController::setCurrentBasemap(Basemap* basemap)
   {
-    if (basemap == m_currentBasemap)
-      return;
-
-    m_currentBasemap = basemap;
-    emit currentBasemapChanged();
+      auto connection = new QMetaObject::Connection();
+
+      auto apply = [basemap, this, connection](Error e){
+        if(e.isEmpty()){
+            if (basemap == m_currentBasemap)
+                  return;
+            if(!basemapMatchesCurrentSpatialReference(basemap)){
```

Comment:
🙃  Has the formatter been run on this? Our coding conventions are always a newline between the if-statement and the brace.